src/validations.rs

```diff
@@ -1,29 +1,41 @@
+use crate::error::ApiError;
+
+const MIN_CARD_NUMBER_LENGTH: usize = 12;
+
 ///
 /// Maximum limit of a card number will not exceed 19 by ISO standards
 ///
-const CARD_NUMBER_LENGTH: usize = 20;
+const MAX_CARD_NUMBER_LENGTH: usize = 19;
 
 #[allow(clippy::expect_used)]
 ///
 /// # Panics
 ///
 /// Never, as a single character will never be greater than 10, or `u8`
 ///
-pub fn luhn_on_string(number: &str) -> bool {
-    let data: Vec<u8> = number
+pub fn luhn_on_string(number: &str) -> Result<bool, ApiError> {
+    let number = number.split_whitespace().collect::<String>();
+
+    let data = number
         .chars()
-        .filter_map(|value| value.to_digit(10))
-        .map(|value| {
-            value
-                .try_into()
-                .expect("error while converting a single character to u8")
-        }) // safety, a single character will never be greater
-        // `u8`
-        .collect();
+        .try_fold(Vec::with_capacity(20), |mut data, character| {
```

Comment:
We can maybe use the `MAX_CARD_NUMBER_LENGTH` const for specifying the capacity?

```suggestion
        .try_fold(Vec::with_capacity(MAX_CARD_NUMBER_LENGTH), |mut data, character| {
```